R/utils.R

```diff
@@ -1634,7 +1634,10 @@ log_err = function(msg) {
 #' @keywords internal
 check_matrix = function(count_mat) {
     if ('matrix' %in% class(count_mat)) {
-        count_mat <- as(Matrix(count_mat, sparse=TRUE), "dgCMatrix")
+        if (typeof(count_mat) != 'integer') {
+            log_err("count_mat is not of integer type")
```

Comment:
I'd prefer a more informative error. 

The logic from Teng is to check if it's a matrix of dgCMatrix. If It's a matrix, it should be integer. 

The error logs should point to the function and reason for users to debug. 

@igordot

Reply:
>  If It's a matrix, it should be integer.

A matrix can also be a `double`. For example, see `typeof(0)` vs `typeof(0L)`.